sklearn/preprocessing/data.py

```diff
@@ -430,6 +440,124 @@ def inverse_transform(self, X, copy=None):
         return X
 
 
+class MaxAbsScaler(BaseEstimator, TransformerMixin):
+    """Scale each feature by its maximum absolute value.
+
+    This estimator scales and translates each feature individually such
+    that the maximal absolute value of each feature in the
+    training set will be 1.0. It does not shift/center the data, and
+    thus does not destroy any sparsity.
+
+    This scaler can also be applied to sparse CSR or CSC matrices.
+
+    Parameters
+    ----------
+    copy : boolean, optional, default is True
+        Set to False to perform inplace scaling and avoid a copy (if the input
+        is already a numpy array).
+
+    Attributes
+    ----------
+    `scale_` : ndarray, shape (n_features,)
```

Comment:
3893cc3239c72096de methinks (July 2014), see 70f520c

On 12 June 2015 at 01:26, Andreas Mueller <notifications@github.com> wrote:

> In sklearn/preprocessing/data.py
> <https://github.com/scikit-learn/scikit-learn/pull/4828#discussion_r32232649>
> :
>
> > +    This estimator scales and translates each feature individually such
> > +    that the maximal absolute value of each feature in the
> > +    training set will be 1.0. It does not shift/center the data, and
> > +    thus does not destroy any sparsity.
> > +
> > +    This scaler can also be applied to sparse CSR or CSC matrices.
> > +
> > +    Parameters
> > +    ----------
> > +    copy : boolean, optional, default is True
> > +        Set to False to perform inplace scaling and avoid a copy (if the input
> > +        is already a numpy array).
> > +
> > +    Attributes
> > +    ----------
> > +    `scale_` : ndarray, shape (n_features,)
>
> On 06/11/2015 09:11 AM, jnothman wrote: backticks no longer needed
> Since when?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/4828/files#r32232649>.
>